sodium.go

```diff
@@ -0,0 +1,83 @@
+// +build compat_test
+
+package secretstream
+
+// #cgo LDFLAGS: -lsodium
+// #include <sodium.h>
+// #include <sodium/crypto_secretstream_xchacha20poly1305.h>
+import "C"
+import (
+	"errors"
+)
+
+type sodiumStream struct {
+	state C.crypto_secretstream_xchacha20poly1305_state
+}
+
+func NewSodiumSendStream(key []byte) (Encryptor,[]byte, error) {
+	res := &sodiumStream{}
+
+	header := C.malloc(C.crypto_secretstream_xchacha20poly1305_HEADERBYTES)
+	rc := C.crypto_secretstream_xchacha20poly1305_init_push(&res.state, (*C.uchar)(header), (*C.uchar)(C.CBytes(key)))
```

Comment:
does `C.CBytes(key)` not need a corresponding `C.free`?

Reply:
yes, fixed